WordPress/Classes/ViewRelated/Site Creation/Shared/TableViewOffsetCoordinator.swift

```diff
@@ -102,16 +127,73 @@ final class TableViewOffsetCoordinator {
 
         let keyboardScreenFrame = payload.frameEnd
         keyboardContentOffset = keyboardScreenFrame.height
+
+        adjustToolbarOffsetIfNeeded()
+    }
+
+    @objc
+    private func keyboardWillHide(_ notification: Foundation.Notification) {
+        toolbarBottomConstraint?.constant = 0
+    }
+
+    private func adjustToolbarOffsetIfNeeded() {
+        guard let toolbar = toolbar, let container = container else {
+            return
+        }
+
+        var constraintConstant = keyboardContentOffset
+
+        if #available(iOS 11.0, *) {
+            let bottomInset = container.safeAreaInsets.bottom
+            constraintConstant -= bottomInset
+        }
+
+        if let header = tableView?.tableHeaderView as? TitleSubtitleTextfieldHeader {
+            let textFieldFrame = header.textField.frame
+
+            let newToolbarFrame = toolbar.frame.offsetBy(dx: 0.0, dy: -1 * constraintConstant)
+
+            toolbarBottomConstraint?.constant = constraintConstant
+            container.setNeedsUpdateConstraints()
+
+            UIView.animate(withDuration: Constants.headerAnimationDuration, delay: 0, options: .beginFromCurrentState, animations: { [weak self] in
+                guard let self = self, let tableView = self.tableView else {
+                    return
+                }
+
+                if textFieldFrame.intersects(newToolbarFrame) {
+                    let contentInsets = UIEdgeInsets(top: -1 * toolbar.frame.height, left: 0.0, bottom: constraintConstant + toolbar.frame.height, right: 0.0)
+                    self.toolbarHasBeenAdjusted = true
+                    tableView.contentInset = contentInsets
+                    tableView.scrollIndicatorInsets = contentInsets
+                }
+                container.layoutIfNeeded()
+                }, completion: { [weak self] _ in
+                    self?.tableViewHasBeenAdjusted = false
+            })
+        }
     }
 
     func startListeningToKeyboardNotifications() {
         NotificationCenter.default.addObserver(self,
                                                selector: #selector(keyboardWillShow),
                                                name: UIResponder.keyboardWillShowNotification,
                                                object: nil)
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(keyboardWillHide),
+                                               name: UIResponder.keyboardWillHideNotification,
+                                               object: nil)
     }
 
     func stopListeningToKeyboardNotifications() {
```

Comment:
I know you didn't change this, but how would feel about removing this method since it's not needed? I'm happy to enqueue a separate PR if you'd prefer.

Reply:
I did change it. It seems necessary to me, as we need to push the Skip button down again when the keyboard is dismissed, or am I missing something?